### PR TITLE
Tag: align with GDS

### LINKED ISSUE #196: Tag: Fix UI Mismatches
GDS: https://share.goabstract.com/f57059a4-edfc-4634-a1e6-c2b586b2eacf?sha=

### PR DESCRIPTION
**Pull request by rosslo**
GDS Tag:
https://app.abstract.com/share/f57059a4-edfc-4634-a1e6-c2b586b2eacf?sha=

# What changed
- remove useless `theme` prop
- add `variant` prop to set the Tag's type (Basic Tag, Button Tag, Action Tag)
- add `action` prop to set the Action Tag's type
- add `icon` prop to set the Button Tag's right icon
- adjust the clickable area of`Tag`
- adjust the layout of `Tag` story

### CODE REVIEW

File: src/General/Tag/Tag.tsx [hunk 1]
```diff
@@ -1,38 +1,65 @@
...
   outline: false,
+  border: 'solid',
 };
 
 interface Props extends React.ComponentPropsWithoutRef<typeof TagContainer> {
   children: React.ReactNode;
-  theme?: string;
+  onClick?(e: React.MouseEvent<HTMLDivElement, MouseEvent>): void;
+  startIcon?: React.ReactNode;
+  endIcon?: React.ReactNode;
```
Comment (tingtingdisco) [inline src/General/Tag/Tag.tsx:62]: why need startIcon and endIcon? they can be in the `children`?

Comment (tingtingdisco) [reply src/General/Tag/Tag.tsx:62]: ah I see the spec

**Comment by tingtingdisco** [comment]
I don't see this is being implemented
Screen Shot 2020-05-06 at 1 43 43 PM

File: stories/General/TagStory.tsx [hunk 2]
```diff
@@ -1,81 +1,294 @@
...
+    fill: ${Greyscale.black};
+  }
+\`;
+<Tag endIcon={<ClickableCloseIcon onClick={() => alert('Clicked!')} />}/>}>Add</Tag>
+`}
+      >
+        <Heading>Customized Tags</Heading>
+        <TagContainer>
+          <Tag
+            endIcon={<ClickableCloseIcon onClick={() => alert('Clicked!')} />}
```
Comment (tingtingdisco) [inline stories/General/TagStory.tsx]: is it possible to make the tag have `onStartIconClick` and `onEndIconClick`, so we don't need to style `ClickableCloseIcon`

File: src/General/Tag/Tag.tsx (hunk 1 above)
Comment (tingtingdisco) [inline src/General/Tag/Tag.tsx]: shall we let users pass any startIcon  and endIcon? It seems like only 3 icons are in the design system
Screen Shot 2020-05-06 at 1 53 58 PM

File: stories/General/TagStory.tsx [hunk 3]
```diff
@@ -1,81 +1,294 @@
...
+      </TagContainer>
+    </StorybookComponent>
+  );
+};
+
+const ClickableTagStory = () => {
+  const usage = `<Tag>Non-clickable tag</Tag>
+<Tag startIcon={<AddIcon onClick={() => alert('Clicked!')} />}>Clickable icon</Tag>
+<Tag endIcon={<CloseIcon onClick={() => alert('Clicked!')} />}>Clickable icon</Tag>
+<Tag onClick={() => alert('Clicked!')}>Clickable tag</Tag>`;
```
Comment (tingtingdisco) [inline stories/General/TagStory.tsx]: I don't see this type of clickable tag in the spec, only cursor becomes pointer when mouse is hovering?

File: stories/General/TagStory.tsx [hunk 4]
```diff
@@ -1,81 +1,294 @@
...
+      <TagContainer>
+        <Tag border="none">None border</Tag>
+      </TagContainer>
+    </StorybookComponent>
+  );
+};
+
+const ClickableTagStory = () => {
+  const usage = `<Tag>Non-clickable tag</Tag>
+<Tag startIcon={<AddIcon onClick={() => alert('Clicked!')} />}>Clickable icon</Tag>
```
Comment (tingtingdisco) [inline stories/General/TagStory.tsx]: I don't see this type of only startIcon can be clicked in the spec.

File: stories/General/TagStory.tsx (hunk 2 above)
Comment (rosslo) [reply stories/General/TagStory.tsx]: The `ClickableCloseIcon` is used to customize the icon color, it's not related to the click event.

Comment (tingtingdisco) [reply stories/General/TagStory.tsx]: yes I mean we should not let user customize icon color

File: stories/General/TagStory.tsx (hunk 3 above)
Comment (rosslo) [reply stories/General/TagStory.tsx]: Those are the example of how to make the Tag be clickable. 
1. add onClick event on the icon you pass, then only the icon will be clickable.
2.add onClick event on the Tag directly, then the whole tag will be clickable.

**Comment by rosslo** [comment]
> I don't see this is being implemented
> Screen Shot 2020-05-06 at 1 43 43 PM

Do you think we should have a `variant` prop for them?

File: stories/General/TagStory.tsx (hunk 4 above)
Comment (rosslo) [reply stories/General/TagStory.tsx]: I just want to make Tag be flexible. Do you think we should limit this?

File: src/General/Tag/Tag.tsx (hunk 1 above)
Comment (rosslo) [reply src/General/Tag/Tag.tsx]: same as https://github.com/glints-dev/glints-aries/pull/323#issuecomment-624464171

File: stories/General/TagStory.tsx (hunk 2 above)
Comment (rosslo) [reply stories/General/TagStory.tsx]: If we agree that we should  a variant prop, we can define it in aries.

**Review by tingtingdisco — APPROVED** [review]
LGTM, nice storybook refactoring, thank u @rosslo!

**Pull request merged**